### PR TITLE
HW8

### PR DESCRIPTION
**Pull request by SvetlanaMag**

### CODE REVIEW

File: HW8/task2.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+printEnterprise(enterprises);
+
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+function getEnterpriseName(arg) {
```
Comment (anatoly-karpovich) [inline HW8/task2.js]: arg - is not a good name. Create names that tell what the variable is, like departmentIdOrName

File: HW8/task2.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+	const arr = []
+	for (let i = 0; i < enterprises.length; i++) {
+		arr.push(enterprises[i].id)
+		arr.push(...enterprises[i].departments.map(el => el.id))
+	}
+	return Math.max(...arr)
+}
+
+function addEnterprise(name) {
+	let nextID = findMaxID() + 1;
```
Comment (anatoly-karpovich) [inline HW8/task2.js]: better generate new id in this function, e.g. findNewId() where you find the max one and add 1

File: HW8/task2.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	const copyEnterprise = structuredClone(enterprises)
```
Comment (anatoly-karpovich) [inline HW8/task2.js]: what for are you making a copy?

File: HW8/task2.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+console.log(addEnterprise('Новое предприятие'));
+
+
+// 4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+// Пример:
+// addDepartment(1, "Название нового отдела")
+
+function addDepartment(id, name) {
+	let nextID = findMaxID() + 1;
+	let copyEnterprise = structuredClone(enterprises)
```
Comment (anatoly-karpovich) [inline HW8/task2.js]: no need to copy, you should change the existing company

**Review by anatoly-karpovich — APPROVED** [review]

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**